qrest/resource.py

```diff
@@ -394,6 +396,24 @@ def check(self, **kwargs):
                         parameter)
                     )
 
+        # ----------------------------------
+        # Check body parameters with their schemas
+        for parameter in kwargs:
+            if parameter not in self.config.parameters:
+                continue
+            if self.config.parameters[parameter].call_location != 'body':
+                continue
+            schema = self.config.parameters[parameter].schema
```

Comment:
It's unfortunate that an extension to `ParameterConfig` requires a change to `Resource.check`. If `ParameterConfig` would have a method like `validate_value`, method `Resource.check` could just ask the original `[Query|Body|File]Parameter` to do the validation. In that case, a change to `ParameterConfig` wouldn't have to ripple through to `ParameterConfig`.

By the way, it might be that I'm oversimplifying things here a bit and overlooking a crucial detail that prohibits the approach outlined above. Also, this goes beyond the scope of this pull request.

Reply:
Agreed.